src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1097,7 +1100,7 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
     if (total_device_input_mem_size > _engine.get_device_info().max_global_mem_size)
         usm_device_allocatable = false;
 
-    bool memory_reuse_by_user = !user_requesting_mem_reuse_false(_node);
+    bool memory_reuse_by_user = _node.is_dynamic() ? !reset : !user_requesting_mem_reuse_false(_node);
```

Comment:
is_dynamic checks both input and output, though allocation need consider only output at each time. So I think we only check is_dynamic_output_layout().

Reply:
I replaced it with `is_dynamic_output_layout()` as you reviewed. Thank you.